### ISSUE TITLE
Test failure

### ISSUE THREAD

**Issue by wxsBSD** [PUNCH-Cyber/stoq#22]
While working on bringing 0.16.0 into our infrastructure I'm getting test failures:

```
======================================================================
FAIL: test_get_invalid_url (core_tests.StoqCoreTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/engshare/third-party2/stoq/0.16.0/src/build-gcc-5-glibc-2.23/build/stoq/tests/core_tests.py", line 95, in test_get_invalid_url
    self.assertIsNone(data)
AssertionError: b'No answer in MultiQuery for 8f15bd05-6bd2-4ed6-9879-03eb416f3410' is not None

----------------------------------------------------------------------
Ran 99 tests in 2.471s

FAILED (failures=1)
Test failed: <unittest.runner.TextTestResult run=99 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=99 errors=0 failures=1>
```

When fetching a non-existent URL our proxy returns a 502 with that string as the contents. I think we should be returning None at https://github.com/PUNCH-Cyber/stoq/blob/master/stoq/core.py#L315, and in all the other places with this pattern like post_file() and put_file().

**Comment by mlaferrera** [comment]
The rationale for returning `response.content` vs `None` is so the plugin can determine whether it wants to do something with the error/response from the server. As an over simplified example, some API's return something similar to `{"err": "Permission denied"}`.

Let me think on it and make sure changing the return to `None` doesn't break anything inadvertently. I don't believe any public (or private plugins that I am aware of) will break with this change, so I don't believe it's a huge deal.

**Comment by wxsBSD** [comment]
Yup, that's part of why I wasn't sure here. It will certainly break things but having a way to determine the response code along with the content in the caller might be a good idea.

```
response = stoq.get_file('http://foo.bar.com')
if response.status_code != 200:
    log(response.content)
    return
```

That will certainly break a lot of things, but would at least make it easier to distinguish.

**Comment by mlaferrera** [comment]
Because `get_file()` accepts both a URL and a file path, returning `response` will add a lot of logic for each `get_file()` request to determine if it is a response object or file content. I'm leaning towards your original recommendation which is simply add `return None` if `raise_for_status()` raises an `HTTPError` exception. There are several plugins that automatically try to load the response as json anyway, which in those cases will raise an exception anyway. I'll finish going through the rest of the plugins to verify, but it seems like that is the best path forward.

**Comment by mlaferrera** [comment]
Merged into v0.17.0
